### ISSUE TITLE
Checkboxes in filter sidebar should be wrapped in label tag

### ISSUE THREAD

**Issue by harrislapiroff** [freedomofpress/pressfreedomtracker.us#2633]
Currently they're only wrapped in a div

<img width="311" height="193" alt="Image" src="https://github.com/user-attachments/assets/ba8b732c-12ab-4593-a109-63ad5dd01ff8" />
